File: engine/engine-model/src/main/scala/tools/packet/ui/GetClockTime.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetClockTime(private var _hour: Int, private var _minute: Int, private var _second: Int) extends PacketInput {
  def hour: Int = _hour

  def minute: Int = _minute

  def second: Int = _second

  override def opcode(): SendOpcode = SendOpcode.CLOCK
}

File: engine/engine-model/src/main/java/tools/packet/ui/GetClockTime.java
package tools.packet.ui;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetClockTime(Integer hour, Integer minute, Integer second) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CLOCK;
   }
}